virtaal/views/langview.py

```diff
@@ -103,19 +102,25 @@ def _get_display_string(self, srclang, tgtlang):
 
     def notify_same_langs(self):
         def notify():
-            for s in [gtk.STATE_ACTIVE, gtk.STATE_NORMAL, gtk.STATE_PRELIGHT, gtk.STATE_SELECTED]:
-                self.popupbutton.child.modify_fg(s, gtk.gdk.color_parse('#f66'))
-        gobject.idle_add(notify)
+            for s in [Gtk.StateType.ACTIVE, Gtk.StateType.NORMAL, Gtk.StateType.PRELIGHT, Gtk.StateType.SELECTED]:
+                self.popupbutton.get_child().modify_fg(s, Gdk.color_parse('#f66'))
+
+        GObject.idle_add(notify)
 
     def notify_diff_langs(self):
         def notify():
+            states = [Gtk.StateFlags.ACTIVE, Gtk.StateFlags.NORMAL, Gtk.StateFlags.PRELIGHT, Gtk.StateFlags.SELECTED]
             if hasattr(self, 'popup_default_fg'):
                 fgcol = self.popup_default_fg
             else:
-                fgcol = gtk.widget_get_default_style().fg
-            for s in [gtk.STATE_ACTIVE, gtk.STATE_NORMAL, gtk.STATE_PRELIGHT, gtk.STATE_SELECTED]:
-                self.popupbutton.child.modify_fg(s, fgcol[s])
-        gobject.idle_add(notify)
+                default_style_context = Gtk.StyleContext()
+                fgcol = {}
```

Comment:
Please check for consistency with the line above: `fgcol = self.popup_default_fg` - there are incompatible types for the call to `.override_color()` below. It could be that using .popup_default_fg is not a clean way, then we can rather get things directly from some style_context.

Your change looks like it should be good, but I can't get it to work correctly. The red foreground colour seems to disappear after some interaction with the "English->English" button. Can you maybe have a look at that?